ocs_ci/ocs/platform_nodes.py

```diff
@@ -2420,3 +2420,62 @@ def restart_nodes_by_stop_and_start_teardown(self):
                 cmd = f"ibmcloud ks worker reboot --cluster {cluster_id} --worker {not_ready_node} -f"
                 out = run_cmd(cmd)
                 logger.info(f"Node restart command output: {out}")
+
+    def create_nodes(self, node_conf, node_type, num_nodes):
+        """
+        Create node on IBM Cloud.
+
+        Args:
+            node_conf (dict): of node configuration
+            node_type (str): type of node to be created
+            num_nodes (int): Number of node instances to be created
+
+        Returns:
+           list: of IBMCloudNode objects
+
+        """
+        logger.info("creating new node")
+        login()
+
+        worker_nodes = get_nodes(node_type="worker")
+        provider_id = worker_nodes[0].get()["spec"]["providerID"]
+        cluster_id = provider_id.split("/")[5]
+
+        cmd = f"ibmcloud ks workers --cluster {cluster_id} --output json"
+        out = run_cmd(cmd)
+        worker_nodes = json.loads(out)
+
+        no_of_nodes = len(worker_nodes) + int(num_nodes)
+        logger.info(f"number of nodes going to be add are : {num_nodes}")
+
+        cmd = (
+            f"ibmcloud ks worker-pool resize --cluster {cluster_id} --worker-pool default"
+            f"  --size-per-zone {no_of_nodes}"
+        )
+        run_cmd(cmd)
+
+        logger.info("Waiting for 600 seconds for node to ready")
+        time.sleep(600)
+
+        worker_nodes = get_nodes(node_type="worker")
+        worker_nodes_not_ready = []
+        for worker_node in worker_nodes:
+            node_id = worker_node.get()["metadata"]["labels"][
+                "ibm-cloud.kubernetes.io/worker-id"
+            ]
+            logger.info(f"{node_id} status is : {worker_node.status()}")
+            if worker_node.status() != "Ready":
+                worker_nodes_not_ready.append(node_id)
+
+        if len(worker_nodes_not_ready) > 0:
```

Comment:
you should also check the number of the nodes which is before calling of this function - store it in variable - then you should check the new number of nodes is equal to number_of_nodes from start of the function + num_nodes.  If that's not he case - raise the exception.

Reply:
Done